clarity/src/vm/analysis/type_checker/mod.rs

```diff
@@ -296,9 +301,85 @@ impl FunctionType {
         }
     }
 
+    /// Returns the type of `value`, after converting any contract principal
+    /// types to callable types. In an initial transaction, arguments are typed
+    /// as contract principals, but they must be principal literals, so they
+    /// may be used to call into a contract.
+    pub fn principal_to_callable_type(
```

Comment:
Minor refactor suggestion:  It looks like this method is a no-op for Clarity 1.  All the inner matches seem to be guarded by checks for Clarity 2.  In that case, could this method instead be factored as follows?

```
pub fn principal_to_callable_type(&self, value: &Value, depth: u8, clarity_version: ClarityVerison) -> TypeResult {
   if clarity_version >= ClarityVersion::Clarity2 {
      self.inner_principal_to_callable_type_clarity2(value, depth)
   }
   else {
      // no-op in Clarity 1
      Ok(())
    }
}

/// Returns the type of `value`, after converting any contract principal
/// types to callable types. In an initial transaction, arguments are typed
/// as contract principals, but they must be principal literals, so they
/// may be used to call into a contract.
fn inner_principal_to_callable_type_clarity2(&self, value: &Value, depth: u8) -> TypeResult {
   /* the code you have, but without the gated Clarity version checks */
}
```

What do you think?

Reply:
The first case in the match, turning the principal literal into a callable, is for both Clarity1 and Clarity2. I still think that could be easier to follow if I do as you recommended. I'll make that change.

Reply:
Refactored in f660448fb